scammer-label-propagation/src/agent.py

```diff
@@ -0,0 +1,149 @@
+    else:
+        for row_idx in range(attackers_df.shape[0]):
+            attacker_info = attackers_df.iloc[row_idx]
+            logger.info(f'{central_node}:\tNew attacker info: {attacker_info}')
+            label_dict = {
+                'entity': attacker_info.name,
+                'label': 'scammer-eoa',
+                'confidence': attacker_info['n_predicted_attacker']/10 * attacker_info['mean_probs_attacker'],
+                'entity_type': forta_agent.EntityType.Address,
+                'metadata': {
+                    'central_node': central_node,
+                    'central_node_alert_id': alert_id,
+                }
+            }
+            finding_dict['labels'] = [forta_agent.Label(label_dict)]
+            all_findings_list.append(Finding(finding_dict))
+    return all_findings_list
+        
+        
+def initialize():
+    global executor
+    executor = ThreadPoolExecutor(max_workers=N_WORKERS)
+    global addresses_analyzed
+    addresses_analyzed = []
+    global global_futures
+    global_futures = {}
+    global global_alerts
+    global_alerts = []
+
+    subscription_json = []
+    for bot in attacker_bots:
+        subscription_json.append({"botId": bot, "chainId": CHAIN_ID})
+    alert_config = {"alertConfig": {"subscriptions": subscription_json}}
+    logger.info(f"Initializing scammer label propagation bot. Subscribed to bots successfully: {alert_config}")
+    return alert_config
+
+
+def provide_handle_alert(w3):
+    logger.debug("provide_handle_alert called")
+    
+
+    def handle_alert(alert_event: forta_agent.alert_event.AlertEvent) -> list:
+        logger.debug("handle_alert inner called")
+        logger.debug(f"AlertId:{alert_event.alert_id};\tName:{alert_event.name};\tAlertHash:{alert_event.hash}")
+        t = time.time()
+        global executor
+        global addresses_analyzed
+        global global_futures
+
+        list_of_addresses = []
+        if alert_event.alert.alert_id == 'SCAM-DETECTOR-ADDRESS-POISONING':
+            logger.info(f"Address poisoning. Addresses: {';'.join([label.entity for label in alert_event.alert.labels])}")
+            return []
+        for label in alert_event.alert.labels:
+            if label.confidence >= ATTACKER_CONFIDENCE and label.entity_type == forta_agent.EntityType.Address:
+                list_of_addresses.append(label.entity)
+        list_of_addresses = list(set(list_of_addresses))
+        for address in list_of_addresses:
+            if address not in addresses_analyzed:
+                logger.debug(f"Adding address {address} to the pool")
+                global_futures[address] = executor.submit(run_all_extended, address, alert_event.alert.alert_id)
+                addresses_analyzed.append(address)
+        logger.info(f"Alert {alert_event.alert.alert_id}:\t{time.time() - t:.10f} s. {len(list_of_addresses)} addresses: {';'.join(list_of_addresses)}")
+        return []
```

Comment:
you are returning no findings here, but do it in handle Block. Ideally you would want the findings returned from handleAlert. This way the alert you generate is tied to the alert that was passed into this function.